DataFormats/L1TCalorimeterPhase2/interface/CaloCrystalCluster.h

```diff
@@ -0,0 +1,152 @@
+          e2x2_(e2x2),
+          e2x5_(e2x5),
+          e3x5_(e3x5),
+          e5x5_(e5x5),
+          standaloneWP_(standaloneWP),
+          electronWP98_(electronWP98),
+          photonWP80_(photonWP80),
+          electronWP90_(electronWP90),
+          looseL1TkMatchWP_(looseL1TkMatchWP),
+          stage2effMatch_(stage2effMatch){};
+
+    virtual ~CaloCrystalCluster(){};
+    inline float calibratedPt() const { return calibratedPt_; };
+    inline float hovere() const { return hovere_; };
+    inline float isolation() const { return iso_; };
+    inline float PUcorrPt() const { return PUcorrPt_; };
+    inline float bremStrength() const { return bremStrength_; };
+    inline DetId seedCrystal() const { return seedCrystal_; };
+    void SetCrystalPtInfo(std::vector<float> info) {
+      std::sort(info.begin(), info.end());
+      std::reverse(info.begin(), info.end());
+      crystalPt_ = std::move(info);
+    };
+    void SetExperimentalParams(const std::map<std::string, float> &params) { experimentalParams_ = params; };
+    const std::map<std::string, float> &getExperimentalParams() const { return experimentalParams_; };
+    inline float experimentalParam(std::string name) const {
+      auto iter = experimentalParams_.find(name);
+      if (iter != experimentalParams_.end()) {
+        return iter->second;
+      } else {
+        edm::LogWarning("CaloCrystalCluster") << "Error: no mapping for ExperimentalParam: " << name << std::endl;
+        return -99.;
+      }
+    };
+
+    inline float e2x2() const { return e2x2_; };
+    inline float e2x5() const { return e2x5_; };
+    inline float e3x5() const { return e3x5_; };
+    inline float e5x5() const { return e5x5_; };
+    inline float standaloneWP() const { return standaloneWP_; };
+    inline float electronWP98() const { return electronWP98_; };
+    inline float photonWP80() const { return photonWP80_; };
+    inline float electronWP90() const { return electronWP90_; };
+    inline float looseL1TkMatchWP() const { return looseL1TkMatchWP_; };
+    inline float stage2effMatch() const { return stage2effMatch_; };
+
+    // The index range depends on the algorithm eta,phi window, currently 3x5
+    // The pt should always be ordered.
+    inline float crystalPt(unsigned int index) const { return (index < crystalPt_.size()) ? crystalPt_[index] : 0.; };
+
+  private:
+    // pT calibrated to Stage-2 (Phase-I) L1EG Objects.  NOTE
+    // all working points are defined with respect to cluster.pt(),
+    // not cluster.calibratedPt()
+    float calibratedPt_;
+    // HCal energy in region behind cluster (for size, look in producer) / ECal energy in cluster
+    float hovere_;
+    // ECal isolation (for outer window size, again look in producer)
+    float iso_;
+    // DetId of seed crystal used to make cluster (could be EBDetId or EEDetId)
+    DetId seedCrystal_;
+    // Pileup-corrected energy deposit, not studied carefully yet, don't use
+    float PUcorrPt_;
```

Comment:
2.9 in http://cms-sw.github.io/cms_coding_rules.html
`Start data member names with lower case. A trailing “_” is the preferred method to distinguish a data member from the getter method (e.g. momentum_).`